website/docs/migration/from_state_notifier.mdx

```diff
@@ -0,0 +1,386 @@
+This enables us to write much simpler logic when handling asynchronous mutations.
+
+## Explicit `.family` and `.autoDispose` modifications
+
+When a `StateNotifier` had family parameters, these were not found into its class, but they were
+defined in its provider; also, and chances are that these parameters must have been piped down its constructor, too.
+
+`Notifier`, instead, has its own `.family` and `.autoDispose` counterparts, such as `FamilyNotifier`
+and `AutoDisposeNotifier`.  
+Like any other provider, these can be combined (aka `AutoDisposeFamilyNotifier`).  
+`AsyncNotifer` has its asynchronous equivalent, too (e.g. `AutoDisposeFamilyAsyncNotifier`).
+
+The main difference (and advantage) is that modifications and any parameters are explicitly stated inside the class;
+furthermore, they are directly injected in the `build` method, so that they're available to the initialization logic.  
+This should bring better readability, more conciseness and less mistakes.
+
+Take the following example, in which a `StateNotifierProvider.family` is being defined.
+
+```dart
+class BugsEncounteredNotifier extends StateNotifier<AsyncValue<int>> {
+  BugsEncounteredNotifier({
+    required this.ref,
+    required this.featureId,
+  }) : super(const AsyncData(99));
+  final String featureId;
+  final Ref ref;
+
+  Future<void> fix(int amount) async {
+    state = await AsyncValue.guard(() async {
+      final old = state.requireValue;
+      final result = await ref.read(taskTrackerProvider).fix(id: featureId, fixed: amount);
+      return max(old - result, 0);
+    });
+  }
+}
+
+final bugsEncounteredNotifierProvider =
+    StateNotifierProvider.family.autoDispose<BugsEncounteredNotifier, int, String>((ref, id) {
+  return BugsEncounteredNotifier(ref: ref, featureId: id);
+});
+```
+
+The main gotcha here is that reading `BugsEncounteredNotifier` is _not_ enough to determine wheter or
+not this `StateNotifier` has modifications (e.g. `.family` and/or `.autoDispose`).  
+To determine that, it is required to read the its provider definition.
+
+Now, let's take a look at its migrated `AsyncNotifier` counterpart:
+
+<AutoSnippet language="dart" {...familyAndDispose}></AutoSnippet>
+
+This last snippet should showcase better readability: it's possible to immediately tell
+what modifications are being set for this Notifier.
+
+## Lifecycles have a different behavior
+
+Lifecycles between `Notifier`/`AsyncNotifier` and `StateNotifier`/`ChangeNotifier` differ substantially.
+
+`StateNotifier` and `ChangeNotifier` instances are indeed bound to their outer
+`StateNotifierProvider` and `ChangeNotifierProvider` providers.  
+When their provider rebuilds, their exposed value, aka their `StateNotifier`/`ChangeNotifier` instance,
+is `disposed`, and then re-created (i.e. its constructor gets invoked again).
+
+This example showcases - again - how the old API have sparse logic:
```

Comment:
I don't think that's something worth mentioning. To begin with, there's not a big difference with the new approach here anyway.